### PR TITLE
Convert full to a team config, add more docs

### PR DESCRIPTION
**Pull request by umairidris**

### CODE REVIEW

File: cmd/tfengine/README.md [hunk 1]
```diff
@@ -6,8 +6,9 @@ The Terraform Engine is a tool to generate complete end-to-end Terraform
 deployments for Google Cloud with security, compliance, and best practices baked
 in.
 
-It introduces the concept of "templates" to wrap Terraform modules. Templates
-define configuration specific to your Google Cloud organization and structure.
+It introduces the concept of "templates". Templates can be used to manage
```
Comment (MartinPetkov) [inline cmd/tfengine/README.md:9]: I don't think it's quite accurate to say templates "manage" TF modules. You use them to generate modules. I suggest rewording "manage".

File: cmd/tfengine/README.md [hunk 2]
```diff
@@ -87,6 +88,33 @@ Use our [example](../../examples/tfengine) configs to quickly get started.
...
+## Defining Architecture
+
+Plan out your org architecture and map them to engine configs.
+Our configs can help you setup a wide variety of architectures that suit your
+needs.
+
+As a rule of thumb, we recommend having one org config that sets up core
+security and compliance infrastructure such as auditing, monitoring and org
+policies. It should define a devops project to manage Terraform
+state and a CICD pipeline.You can also use this config to define the org's
```
Comment (MartinPetkov) [inline cmd/tfengine/README.md:100]: Need space between `.You`

File: examples/tfengine/README.md [hunk 3]
```diff
@@ -2,14 +2,29 @@
...
+
+- [./folder_foundation.hcl](./folder_foundation.hcl): This example is similar to
+  the [./org_foundation.hcl](./org_foundation.hcl) example except it sets
+  everything on a folder instead of an org. This is useful if you want to test
+  the generated configs in an isolated environment before promoting them to org
+  level. This example assumes the root folder has already been created (for
+  example by an org foundation config or other means).
+
+- [./team.hcl](./team.hcl): This example sets up a sample team or application
+  within a folder. This example focusses on projects and concrete resources to
```
Comment (MartinPetkov) [inline examples/tfengine/README.md:26]: Typo in `focusses`

File: examples/tfengine/README.md [hunk 4]
```diff
@@ -2,14 +2,29 @@
...
+  pipelines.
 
-```bash
-# To be run at the git repository root.
-go install ./cmd/tfengine
-tfengine --config_path=examples/tfengine/simple.hcl --output_path=examples/tfengine/generated/simple
-tfengine --config_path=examples/tfengine/full.hcl --output_path=examples/tfengine/generated/full
-```
+- [./org_foundation.hcl](./org_foundation.hcl): This example expands upon the
+  [./devops.hcl](./devops.hcl) example to setup an org with additional security
```
Comment (MartinPetkov) [inline examples/tfengine/README.md:14]: I suggest expanding a bit on what "additional security and compliance" means.

File: examples/tfengine/org_foundation.hcl [hunk 5]
```diff
@@ -84,3 +84,23 @@ template "org_policies" {
...
+    display_name = "prod"
+  }
+}
+
+# Prod folder for team 1.
+template "folder_team1" {
+  recipe_path = "{{$recipes}}/folder.hcl"
+  output_path = "./live/prod/team1"
+  data = {
+    parent_type                  = "folder"
```
Comment (MartinPetkov) [inline examples/tfengine/org_foundation.hcl:102]: Do you need a parent_id here?

Comment (umairidris) [reply examples/tfengine/org_foundation.hcl:102]: The add_parent_folder_dependency below this field should take care of that. Any suggestions on making that more obvious?

Comment (MartinPetkov) [reply examples/tfengine/org_foundation.hcl:102]: I see. I'm not really sure how to make it more clear, let's leave it for now.
